src/Model/GiftCard.php

```diff
@@ -269,4 +273,28 @@ public function getOrigin(): ?string
     {
         return $this->origin;
     }
+
+    public function getValidUntil(): ?DateTimeInterface
+    {
+        return $this->validUntil;
+    }
+
+    public function setValidUntil(?DateTimeInterface $validUntil): void
+    {
+        $this->validUntil = $validUntil;
+    }
+
+    public function isExpired(DateTimeInterface $date = null): bool
```

Comment:
Do we need the argument to this method?

Reply:
for mocking and tests especially. But it also allows testing a specific GC is expired at a specific date. Can be handy :D

Reply:
The test argument is not valid, since you can just initialize the gift card with the date you want to test I guess? :D

I am leaning more and more towards [YAGNI](https://martinfowler.com/bliki/Yagni.html) these days and this is an example of that (though very small)

Reply:
You can initialize the date for GC, but not the local date. And mocking dates is a pain.
Another solution would be to use a service to get the date, but it is not possible inside an entity

Reply:
Ah, yeah, of course. We keep it :D Read it the other way around :|